test/base_test.py

```diff
@@ -77,6 +77,27 @@ def test_data_on_array_with_format(self, server, format_arg):
 
         assert np.array_equal(retrieved_data, data)
 
+    # TODO: What should we do for csv, tiff
```

Comment:
@t20100 :point_up:

Reply:
🤷 What does it currently returns? It doesn't sounds useful anyway

Reply:
The server encounters an error:

- `csv`:
```
  File ".../h5grove/h5grove/fastapi_utils.py", line 113, in get_data
    h5grove_response = encode(data, format)
  File ".../h5grove/h5grove/encoders.py", line 131, in encode
    csv_encode(content_array),
  File ".../h5grove/h5grove/encoders.py", line 58, in csv_encode
    np.savetxt(buffer, data, delimiter=",")
  File ".../h5grove/lib/python3.10/site-packages/numpy/lib/_npyio_impl.py", line 1573, in savetxt
    raise ValueError(
ValueError: Expected 1D or 2D array, got 0D array instead
```
- `tiff`: 
```
  File ".../h5grove/h5grove/encoders.py", line 149, in encode
    tiff_encode(content_array),
  File ".../h5grove/h5grove/encoders.py", line 78, in tiff_encode
    tifffile.imwrite(buffer, data, photometric="minisblack")
  File ".../h5grove/lib/python3.10/site-packages/tifffile/tifffile.py", line 966, in imwrite
    result = tif.write(data, shape, dtype, **kwargs)
  File ".../h5grove/lib/python3.10/site-packages/tifffile/tifffile.py", line 2020, in write
    raise RuntimeError(
RuntimeError: length of storedshape 4 not in (5, 6)
```

In both cases, the server returns an HTTP Error 500 to the user. I wondered if we could not make it more explicit.

Reply:
> I wondered if we could not make it more explicit.

+1

What would be the best error code? [415 Unsupported Media Type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/415) ?

Reply:
We already stuff put in place to raise [422 Unprocessable Content](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) if the data is not numeric: https://github.com/silx-kit/h5grove/blob/53d7d0fdf7d8d89aa0d5297d7183d8ad7bc1975c/h5grove/encoders.py#L124 and https://github.com/silx-kit/h5grove/blob/53d7d0fdf7d8d89aa0d5297d7183d8ad7bc1975c/h5grove/content.py#L297

We could reuse it to raise a 422 error in case of unsupported encoding for scalars.

Reply:
6028d8900020193f14d7bcc9d72f559d5a0f81ef